app/services/claims/fee_calculator/calculate.rb

```diff
@@ -83,20 +84,35 @@ def fee_scheme
         @fee_scheme ||= client.fee_schemes(type: scheme_type, case_date: earliest_representation_order_date.to_s(:db))
       end
 
+      def bill_scenario
+        CCR::CaseTypeAdapter::BILL_SCENARIOS[case_type.fee_type_code.to_sym]
+      end
+
       # TODO: consider creating a mapping to fee calculator id's
       # - less "safe" but faster/negates the need to query the API??
       #
       def scenario
-        # TODO: create select/find_by calls to list endpoints in client gem
+        # TODO: create select/find_by calls to retrieve endpoint data by attribute value
+        # as opposed to being limited to parameter queries
+        # e.g. fee_scheme.scenarios.find_by(code: bill_scenario)
+        #
         fee_scheme.scenarios.select do |s|
-          s.code.eql?(CCR::CaseTypeAdapter::BILL_SCENARIOS[case_type.fee_type_code.to_sym])
+          s.code.eql?(bill_scenario)
         end&.first
       end
 
+      # Send a default offence as fee calc currently requires offences
+      # for some prices even though the values are identical for different
+      # offence classes/bands.
+      # TODO: fee calculator API should not require
+      # offences for at least "Elected case not proceeded"
+      #
       def offence_class
-        # some/all fixed fees do not require offences and they have no bearing on calculated fee amount
-        # TODO: make conditional on fee scheme version
-        offence&.offence_class&.class_letter || offence&.offence_band&.description
+        if agfs_reform?
+          offence&.offence_band&.description || '17.1'
+        else
+          offence&.offence_class&.class_letter || 'H'
+        end
       end
```

Comment:
Something about this seems wrong but I can't put my finger on it. 
Would (but  with a better name)
```ruby
def default_value_if_offence_class_missing
  agfs_reform? ? '17.1' : 'H'
end
```
 be more declarative?

Reply:
will do

Reply:
@colinbruce or could put the default on the offence model?

Reply:
:+1:

As discussed, let's leave as-is for now, this code is pretty volatile at the moment, I can always suggest an improvement... if I can think of it! :)